packages/chain-controller/src/SnapHandlerClient.test.ts

```diff
@@ -0,0 +1,68 @@
+import type { SnapId } from '@metamask/snaps-sdk';
+import { HandlerType } from '@metamask/snaps-utils';
+
+import { SnapHandlerClient } from './SnapHandlerClient';
+
+const snapId = 'local:localhost:3000' as SnapId;
+
+describe('SnapHandlerClient', () => {
+  describe('submitRequest', () => {
+    const method = 'chain_method';
+    const params = {};
+    const request = {
+      snapId,
+      origin: 'metamask',
+      handler: HandlerType.OnRpcRequest,
+      request: {
+        id: expect.any(String),
+        jsonrpc: '2.0',
+        method,
+        params,
+      },
+    };
+    const response = {
+      success: true,
+    };
+
+    it('should call a method and return the result', async () => {
```

Comment:
Following our testing practices, we shouldn't start a test name with `should`.

https://github.com/MetaMask/contributor-docs/blob/main/docs/e2e-testing.md#test-names

```suggestion
    it('returns a result when a method is called', async () => {
```